packages/pangea-node-sdk/src/services/authn.ts

```diff
@@ -0,0 +1,421 @@
+import PangeaResponse from "../response";
+import BaseService from "./base";
+import PangeaConfig from "../config";
+import { AuthN } from "../types";
+
+import { schema } from "../utils/validation";
+/**
+ * AuthnService class provides methods for interacting with the AuthN Service
+ * @extends BaseService
+ */
+class AuthnService extends BaseService {
+  constructor(token: string, config: PangeaConfig) {
+    super("authn", token, config);
+    this.apiVersion = "v1";
+  }
+
+  // authn::/v1/password/update
+  /**
+   * @summary Change a user's password
+   * @description Change a user's password given the current password
+   * @param {String} o.email - An email address
+   * @param {String} o.old_secret - The old password
+   * @param {String} o.new_secret - The new password
+   * @returns {Promise<PangeaResponse<{}>>} - A promise representing an async call to the endpoint
+   * @example
+   * const response = await authn.passwordUpdate(
+   *   "example@example.com",
+   *   "hunter2",
+   *   "My2n+Password"
+   * );
+   */
+  passwordUpdate(o: AuthN.PasswordUpdateRequest): Promise<PangeaResponse<{}>> {
+    if (!schema.string(o.email)) {
+      throw "passwordUpdate was called without supplying an email";
+    }
+    if (!schema.string(o.old_secret)) {
+      throw "passwordUpdate was called without supplying an old_secret";
+    }
+    if (!schema.string(o.new_secret)) {
+      throw "passwordUpdate was called without supplying an new_secret";
+    }
+
+    return this.post("password/update", o);
+  }
+
+  // authn::/v1/user/create
+  /**
+   * @summary Create a user
+   * @description Create a user
+   * @param {String} user.email - An email address
+   * @param {String} user.authenticator - A provider-specific authenticator, such as a password or a social identity
+   * @param {String} user.id_provider - Mechanism for authenticating a user's identity
+   * @param {Boolean} user.verified - True if the user's email has been verified
+   * @param {Boolean} user.require_mfa - True if the user must use MFA during authentication
+   * @param {Object} user.profile - A user profile as a collection of string properties
+   * @param {Array<String>} user.scopes - A list of scopes
+   * @returns {Promise<PangeaResponse<{}>>} - A promise representing an async call to the endpoint
+   * @example
+   * const response = await authn.userCreate({
+   *   email: "example@example.com",
+   *   authenticator: "My1s+Password",
+   *   id_provider: "password",
+   *   verified: false,
+   *   require_mfa: false,
+   *   profile: {
+   *     first_name: "Joe",
+   *     last_name: "User",
+   *   },
+   *   scopes: ["scope1", "scope2"]
+   * });
+   */
+  userCreate(user: AuthN.UserCreateRequest): Promise<PangeaResponse<AuthN.UserCreateResult>> {
```

Comment:
If you made an AuthNUser class that extended BaseService you could add a new property to base class like apiPathPrefix or something that gets injected 

async get(endpoint: string, path: string): Promise<PangeaResponse<any>> {
    const fullpath = `${this.apiVersion}/${this.apiPathPrefix}${path}`;
    return await this.request.get(endpoint, fullpath);
  }

So for AuthN User, the pathPrefix = "user/", kindof idea. Just a suggestion

Reply:
My thought on making user it own class

would be add the property 

get user() {
   return AuthNUser(this.token, this.config)
}

or if we don't want to create an new object per user access then could do

get user() {
  // Where authnUser is created in the AuthN constructor
  return this.authnUser;
}

But I also like the idea of making user a method, and having something like

user({ identity: string, email: string }) {
  return AuthNUser(this.token, this.config, { identity, email });
}

which could then simply further calls made to get the user because the target identity/email was already set. It does change the paradigm where the method is a direct wrapper around an api, although this solution doesn't mean they couldn't still pass the identity/email again in the request.

It could make their code something like
```
const authnClient = AuthN(token, config);

const colinClient = authnClient.user({ email: "colin.fagan@pangea.cloud" });
const colin = colinClient.get();

...

colinClient.update(...);
```

Idk, it doesn't fully translate on like a user.create or user.list, so potentially makes things just more complicated than they need to be, probably just user as a property makes the most sense

Reply:
IMO I like more the idea of methods like API wrappers and user as a property. I don't think we need to create a new AuthNUser each time we want to do a call